### DISCUSSION TITLE
Server Replied "504 Gateway Timeout"

### DISCUSSION THREAD

**Discussion by david-steg** [nextcloud/all-in-one#706] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Upload biggers Files(~1.5Gb) from Nextcloud Desktop 

### Expected behavior 
Upload without Errors

### Actual behavior 
Uploads with errors "Server Replied "504 Gateway Timeout""

### Host OS 
Ubuntu 20.04

#### Nextcloud AIO version 
v1.1.0

#### Current channel 
You are running the [latest](https://github.com/nextcloud/all-in-one#how-to-switch-the-channel) channel.

#### Other valuable info 
Log in the WebUI is empty

**Comment by szaimen** [comment]
How long did the transfer need and are you running aio behind a reverse proxy?

**Comment by david-steg** [comment]
I need to oberve the exact behavior.. But waht I saw, ist that the Upload from the affected file is stopped and marked with a red cross. After the next Nextcloud Desktop boot, it will mark as "sync"(green hack).
Yes, it run behinde a Reverse proxy, pfSense HAProxy.

**Comment by szaimen** [comment]
How long did the upload take?

**Comment by david-steg** [comment]
Hello, I could observe a big file Upload:
I tested two .mkv file, 8.7GB and 11.6GB.

Both had the same behaviour:
- Synchronizing with local folder: 2' - OK
- Upload with 4-18MB/s: 8.7GB(11') | 11.6GB(14') - OK
- Sync: after exact 5' (504 Gateway Timeout to "MOVE /upload/xxx.file), show up juste for a few secondes.
![Capture](https://user-images.githubusercontent.com/71213394/168132862-ee2cc112-a6d7-42e0-abdc-fca4a0dd3369.PNG)
- Some other Sync: ~5'
- Error "423 Lock" to " Move ..." upload.part is locked (Red Cross on the file)
![Screenshot 2022-05-12 191334](https://user-images.githubusercontent.com/71213394/168132854-ee311604-c9c3-41a4-8b40-1831cc33ac8e.png)
- Desktop: All Sync, with Error 423
- After ~3' No Error 423 but with Server Log --> Error: rmdir(/mnt/ncdata/djs/uploads/3417241386): No such file or directory at /var/www/html/lib/private/Files/Storage/Local.php#128

**Comment by szaimen** [comment]
Thanks!

Can you send the apache container logs and the nextcloud container logs as well as the nextcloud logs from directly after the problem happens?

Thank you!

**Comment by szaimen** [comment]
Also, can you send your reverse proxy conf?

The default haproxy timeout may be the issue here...

**Comment by szaimen** [comment]
I just tried on my testinstance without reverse proxy and it successfully uploaded a 7GB file without any timeout in 25min.

**Comment by szaimen** [comment, marked as answer]
Hello @david-steg any update here?

**Comment by szaimen** [comment]
Converting this to a discussion...
